app/models/game.rb

```diff
@@ -29,29 +29,210 @@ def populate_board!
+        # to the specified coordinate set between the piece causing check 
+          unless opponent_piece.is_obstructed? x_of_threat, y_of_threat
+          # If both criteria are met, at least one opponent_piece can block the piece causing check
+            return true
+          end
+        end
+      end
+    end
+    
+  # Return false: no opponent_piece can block the player piece causing check
     false
   end
+  
+  def checked_king_can_escape?
+  # Store opponent_king coordinates locally
+    okx = @okx
+    oky = @oky
+    
+  # Build two-dimensional array of opponent king's surrounding coordinate sets
+  # (leave behind any coordinate sets that contain a 0 or 9 (would be invalid moves))
+    possible_escape_coordinates_for_king = [
+      [okx, oky - 1],
+      [okx, oky + 1],
+      [okx + 1, oky],
+      [okx - 1, oky],
+      [okx - 1, oky - 1],
+      [okx - 1, oky + 1],
+      [okx + 1, oky - 1],
+      [okx + 1, oky + 1]
+    ].delete_if { |coordinate_set| coordinate_set.include? 0 }
+     .delete_if { |coordinate_set| coordinate_set.include? 9 }
+    
+  # Iterate through opponent king's valid surrounding coordinate sets
+    possible_escape_coordinates_for_king.each do |escape_x, escape_y|
+    # Check if it would be valid for the opponent_king to move to the specified surrounding coordinate set
+      if @opponent_king.valid_move? escape_x, escape_y
+      # Check if the opponent_king wouldn't obstructed while attempting to move to the specified surrounding coordinate set
+        unless @opponent_king.is_obstructed? escape_x, escape_y
+        # Check if a player piece doesn't occupy the specified surrounding coordinate set
+          unless @opponent_king.player_piece_occupies_destination? escape_x, escape_y
+          # Check if the opponent_king wouldn't enter check while attempting to move to the specified surroudning coordinate set
+            unless @opponent_king.move_would_cause_check? escape_x, escape_y
+            # If above criteria are met, the opponent_king can escape check
+              return true
+            end
+          end
+        end
+      end
+    end
+
+  # Return false: the opponent_king can not escape check
+    false
+  end
+  
+  def set_remaining_piece_data(color)
+  # Query for remaining player pieces
+    @remaining_player_pieces = pieces.where(game_id: id, color: color, captured?: false)
+  # Query for remaining opponent pieces
+    @remaining_opponent_pieces = pieces.where(game_id: id, captured?: false).where.not(color: color)
+  # Grab the king from @remaining_opponent_pieces
+    @opponent_king = @remaining_opponent_pieces.select { |remaining_opponent_piece| remaining_opponent_piece.type.eql? 'King' }.last
+
+  # Store opponent_king coordinates for shorter reference
+    @okx = @opponent_king.x_coordinate
+    @oky = @opponent_king.y_coordinate
+  end
+  
+  def set_coordinates_between_piece_causing_check_and_opponent_king
```

Comment:
I do agree that the function for checkmating is a complex process but what I was more specifically referring to is the part of your code that defines the positions between two points. It caught my attention because of it's similarity to the code I wrote for 'is_obstructed?' before I refactored it to what it is now. This gave me the idea of making a method that specifically helps us define positions between two points. But perhaps it would be better explained in a PR at a future date.